src/feature_extraction.py

```diff
@@ -69,8 +71,33 @@ def execute_sift_extraction(photo_path, bounding_box, with_colour=1):
     photo = cv2.imread(photo_path, with_colour)
     photo_cropped = photo[bounding_box.y0:bounding_box.y0 + bounding_box.dy,
                     bounding_box.x0:bounding_box.x0 + bounding_box.dx]
-    photo_kp, photo_desc = gen_sift_features(photo_cropped)
-    return photo_cropped, photo_kp, photo_desc
+    return generate_image_descriptors(photo_cropped)
+
+
+def generate_image_descriptors(photo_cropped):
+    datagen = ImageDataGenerator(
+        rotation_range=40,
+        width_shift_range=0.2,
+        height_shift_range=0.2,
+        shear_range=0.2,
+        zoom_range=0.2,
+        horizontal_flip=True,
+        fill_mode='nearest')
+
+    results = list()
+
+    x = img_to_array(photo_cropped)
+    x = x.reshape((1,) + x.shape)
+
+    batch = datagen.flow(x, batch_size=config.data_multiplication_factor)
+    for i in range(0, config.data_multiplication_factor + 1):
+        pic = next(batch)
+        pic = pic.reshape(pic.shape[1:4])
+        pic = pic.astype(np.uint8, copy=False)
+        _, photo_desc = gen_sift_features(pic)
+        results.append(photo_desc)
+
+    return results
 
 
 def generate_subdir_path(dir_path):
```

Comment:
Replace string concatenation (in yield) with os.path.join -> better, more general way to join paths in python (OS independent).

Reply:
done